mlir/lib/Dialect/MQTOpt/Transforms/ToQuantumComputationPattern.cpp

```diff
@@ -62,33 +68,41 @@ struct ToQuantumComputationPattern final : mlir::OpRewritePattern<AllocOp> {
    * variables.
    *
    * @return The index of the qubit in the list of previously defined qubit
-   * variables.
+   * variables. Throws an exception if the qubit is not found.
```

Comment:
Doxygen has a dedicated syntax for documenting the exceptions a function may throw.
It's also always a good idea to denote the particular type of exception so that people know what they should be catching.

```suggestion
   * variables. 
   *
   * @throws `std::runtime_error` if the qubit is not found.
```